Description: Improvements to the Makefile.
 Replace INSTALL_PATH with the most commonly use DESTDIR variable.
 Add an 'install' target.
Author: Alessio Treglia <alessio@debian.org>
Forwarded: http://code.google.com/p/leveldb/issues/detail?id=27

Rebased for v1.20: Ikey Doherty <michael.i.doherty@intel.com>
---
 Makefile |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

--- a/Makefile
+++ b/Makefile
@@ -2,6 +2,10 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file. See the AUTHORS file for names of contributors.
 
+PREFIX ?= /usr
+LIBDIR ?= lib64
+INCLUDEDIR ?= include
+
 #-----------------------------------------------
 # Uncomment exactly one of the lines labelled (A), (B), and (C) below
 # to switch between compilation modes.
@@ -141,6 +145,19 @@ endif  # PLATFORM_SHARED_EXT
 
 all: $(SHARED_LIBS) $(SHARED_PROGRAMS) $(STATIC_OUTDIR)/libleveldb.a $(STATIC_OUTDIR)/libmemenv.a $(STATIC_PROGRAMS)
 
+LIBRARY = $(STATIC_OUTDIR)/libleveldb.a
+install: all
+	install -d $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	install -d $(DESTDIR)$(PREFIX)/$(INCLUDEDIR)/leveldb
+	install -m 0644 $(LIBRARY) $(DESTDIR)$(PREFIX)/$(LIBDIR)
+ifneq ($(PLATFORM_SHARED_EXT),)
+	install -m 0644 $(SHARED_OUTDIR)/$(SHARED_LIB3) $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	cd $(DESTDIR)$(PREFIX)/$(LIBDIR) && ln -sf $(SHARED_LIB3) $(SHARED_LIB2)
+	cd $(DESTDIR)$(PREFIX)/$(LIBDIR) && ln -sf $(SHARED_LIB3) $(SHARED_LIB1)
+endif
+	install -m 0644 include/leveldb/* $(DESTDIR)$(PREFIX)/$(INCLUDEDIR)/leveldb
+
+
 check: $(STATIC_PROGRAMS)
 	for t in $(notdir $(TESTS)); do echo "***** Running $$t"; $(STATIC_OUTDIR)/$$t || exit 1; done
 
